src/modules/search-result/author.jsx

```diff
@@ -1,28 +1,80 @@
-import React from 'react'
+import React, { useState, useEffect, useRef } from 'react'
 
 import styles from './author.css'
 
+import { Button } from 'elements'
 import { classNames } from 'utils'
 
-const List = ({ children, className, ...htmlProps }) => (
-  <ol className={classNames.use(styles.list).join(className)} {...htmlProps}>
-    {children}
-  </ol>
+const Item = ({ name, className, ...htmlProps }) => (
+  <li
+    itemProp="author"
+    itemScope=""
+    itemType="https://schema.org/Person"
+    className={classNames.use(className)}
+    {...htmlProps}
+  >
+    <a
+      href={`https://core.ac.uk/search?q=author:(${name})`}
+      className={styles.link}
+    >
+      {name.replace(',', ' ')}
+    </a>
+  </li>
 )
 
-const Item = ({ children, href, ...htmlProps }) => {
-  const ElementType = href != null ? 'a' : 'span'
+const List = ({ authors, className, ...htmlProps }) => {
+  const showMoreRef = useRef(null)
+  const [isExpanded, setIsExpanded] = useState(false)
+
+  useEffect(() => {
+    if (isExpanded)
+      showMoreRef.current.parentElement.nextElementSibling.firstChild.focus()
```

Comment:
I am afraid this effect will work before the actual click. Let me check how exactly it works... ⏳

Reply:
there is `useLayoutEffect` which I wanted initially use but then it worked even with useEffect so I went with it.  (`useLayoutEffect` is just client side hook and I didn't want to break the SSR)

Reply:
Yes, it does. Once I mouse-down it, the list changes but the click should happen only when the mouse is released.

It's not about the `useEffect` hook, it's the way it's implemented.

Reply:
Alright but I don't think this is a big deal, is it?

Reply:
It's not 🙂